kustomize/commands/internal/util/configmapSecretFlagsAndArgs.go

```diff
@@ -114,6 +137,49 @@ func (a *ConfigMapSecretFlagsAndArgs) ExpandFileSource(fSys filesys.FileSystem)
 	return nil
 }
 
+// UpdateLiteralSources looks for literal sources that already exist and tries
+// to replace their values with new values.
+// The key specified must exist in the target resource (ConfigMap or Secret).
+func UpdateLiteralSources(
+	args *types.GeneratorArgs,
+	flags ConfigMapSecretFlagsAndArgs,
+) error {
+	sources := make(map[string]any)
+	for _, val := range args.LiteralSources {
+		key, value, found := strings.Cut(val, "=")
```

Comment:
Not necessary, but trying to be super careful since we are dealing with strings here, and users can manipulate the input in any way. Do we want to first check the occurrences of "=" to verify it is appearing once, and return early if it is a malformed string?

Reply:
Yes, I think that'd be a good idea.

Reply:
Updated!